classification_analysis.ipynb

```diff
@@ -240,29 +323,57 @@
     "                                                                                  tr['start_ts'],\n",
     "                                                                                  tr['end_ts'])\n",
     "                        \n",
-    "                        if section_gt_leg[\"type\"] == \"WAITING\": \n",
+    "                        if section_gt_leg[\"type\"] == \"WAITING\":\n",
     "                            continue\n",
+    "                        \n",
     "                        gts = {'start_ts': section['start_ts'], \n",
     "                               'end_ts': section['end_ts'], \n",
-    "                               'mode': section_gt_leg['mode']}\n",
+    "                               'mode': section_gt_leg['mode']\n",
+    "                               }\n",
+    "\n",
     "                        tr_gts.append(gts)\n",
+    "\n",
+    "                    gt_location_data = get_gt_location_data(FILE_MAPPING[pv], tr['trip_id_base'])\n",
+    "                    run_id = r['trip_run']\n",
+    "                    print(run_id, tr['trip_id'], tr['trip_id'])\n",
+    "                    gt_location_data = gt_location_data.query(\"source == @os\")\n",
+    "                    gt_location_data = gt_location_data.loc[gt_location_data.run == str(run_id), :].reset_index(drop=True, inplace=False)\n",
+    "                    gt_dfs.append(gt_location_data)\n",
+    "\n",
+    "                print(5*'-')\n",
+    "\n",
     "                # now, we build a timeline for each trip\n",
     "                trip = tr.copy()\n",
     "                trip['ss_timeline']  = tr_ss\n",
     "                trip['gts_timeline'] = tr_gts\n",
+    "                trip['gt_location_df'] = gt_dfs\n",
```

Comment:
I *think* this is what you meant by asking about concatenation. Since you are working at the trip level for the matching, you want to concatenate the section reference trajectories into one giant trip trajectory. That is fine. You can always split it up again by filtering by the `start_ts` and `end_ts` of each `sr`